sdks/python/.pylintrc

```diff
@@ -97,7 +96,6 @@ disable =
   cyclic-import,
   design,
   fixme,
-  function-redefined,
```

Comment:
Besides the warning in `DecoratorsTest`, it throws warnings on some decorators and type hints tests, and also some DoFns and PTransform redefinitions. Seems like this warning could be disabled since the warnings seem to be coherent with what is being tested.

Reply:
I think we should either:
1. Choose a new name for the thing that is redefined, or
2. delete it, or
3. If it's really intentional to redefine it, then just disable the warning for those specific cases. 

I looked through the warnings in the [precommit](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Commit/12100/console):
```
09:32:10 ************* Module apache_beam.typehints.decorators_test
09:32:10 apache_beam/typehints/decorators_test.py:282:2: E0102: method already defined line 131 (function-redefined)
09:32:10 ************* Module apache_beam.typehints.typehints_test
09:32:10 apache_beam/typehints/typehints_test.py:1123:4: E0102: function already defined line 1113 (function-redefined)
09:32:10 apache_beam/typehints/typehints_test.py:1144:4: E0102: function already defined line 1133 (function-redefined)
09:32:10 apache_beam/typehints/typehints_test.py:1176:4: E0102: function already defined line 1163 (function-redefined)
09:32:10 apache_beam/typehints/typehints_test.py:1200:4: E0102: function already defined line 1187 (function-redefined)
09:32:10 ************* Module apache_beam.transforms.userstate_test
09:32:10 apache_beam/transforms/userstate_test.py:357:6: E0102: method already defined line 351 (function-redefined)
09:32:10 ************* Module apache_beam.options.pipeline_options_test
09:32:10 apache_beam/options/pipeline_options_test.py:508:4: E0102: class already defined line 503 (function-redefined)
09:32:10 ************* Module apache_beam.examples.snippets.snippets_test
09:32:10 apache_beam/examples/snippets/snippets_test.py:381:6: E0102: class already defined line 360 (function-redefined)
```

The decorators_test at least seems to be referring to the fact that `test_origin` is redefined. So one of these tests is not running. It seems like there's more to do here

Reply:
Done, I applied your suggestions 👍 , there are no more warnings of this kind.